crates/nargo/src/cli/compile_cmd.rs

```diff
@@ -42,14 +42,23 @@ pub fn generate_circuit_and_witness_to_disk<P: AsRef<Path>>(
     allow_warnings: bool,
 ) -> Result<PathBuf, CliError> {
     let compiled_program = compile_circuit(program_dir.as_ref(), false, allow_warnings)?;
-    let serialized = compiled_program.circuit.to_bytes();
 
-    let mut circuit_path = create_named_dir(circuit_dir.as_ref(), "build");
+    preprocess_with_path(circuit_name, circuit_dir.as_ref(), compiled_program.circuit.clone())?;
+
+    let mut circuit_path = create_named_dir(circuit_dir.as_ref(), "target");
     circuit_path.push(circuit_name);
     circuit_path.set_extension(ACIR_EXT);
+
+    let serialized = compiled_program.circuit.to_bytes();
     let path = write_to_file(serialized.as_slice(), &circuit_path);
     println!("Generated ACIR code into {path}");
 
+    let mut hasher = Sha256::new();
```

Comment:
Probably better to use this method from acvm fro hashing the circuit, so you don't pull in another dependency: https://github.com/noir-lang/acvm/blob/b9091461e199bacdd073cc9b31f03dade0b4fb2d/acvm/src/lib.rs#L198

Reply:
Ah missed that method. Switched to using this method